### DISCUSSION TITLE
Model X1-1.1-S-D trouble

### DISCUSSION THREAD

**Discussion by etf3k** [syssi/esphome-modbus-solax-x1#57] [Q&A] [answered]
I didnt become any answer from my Solax. Must i activate something in the menu ? 

I add my logs with debugger and logger on. 

Setting : Raspi with HomeAssistent -> Esp8266 -> RS485 to TTL Board

[logs_esphome-web-afe0c6_logs.txt](https://github.com/syssi/esphome-solax-x1-mini/files/11860411/logs_esphome-web-afe0c6_logs.txt)

**Comment by syssi** [comment]
It looks like you have change the RX/TX pins to GPIO1/3. Did you see this warning?

```
[19:12:45][W][uart.arduino_esp8266:127]:   You're using the same serial port for logging and the UART component. Please disable logging over the serial port by setting logger->baud_rate to 0.
```

Please disable to serial logging by setting the baud rate of the logger to 0:

```
logger:
  level: DEBUG
  baud_rate: 0
```

or change the config back to GPIO4/5 and attach the RS485 converter to these GPIOs.

**Reply by syssi** [reply]
Are you sure the device is called a X1 Mini and not X1 Boost?

**Reply by etf3k** [reply]
Solax X1-1.1-S-D MINI G3.1 | 1Ph. says the Internet :D 

![grafik](https://github.com/syssi/esphome-solax-x1-mini/assets/95611904/59ba9f1d-eb38-4a11-a0bb-fc37fcdc733d)

The Pinout says Pin 1 and 3 , its a ESP 8266 12-E NodeMCU

I will test the logger baud rate.

**Reply by syssi** [reply]
Please ignore the RX/TX labels. They mean a hardware UART is wired to these pins. If you assign the `uart` component to any other GPIO a implementation called SoftwareSerial is used which handles the communication. In other words: You aren't limited to GPIO1/3 if you are using ESPHome.

**Reply by syssi** [reply]
Could you provide a photo of your RS485 converter? Does the TX LED blink periodically?

**Reply by etf3k** [reply]
![1687767536913](https://github.com/syssi/esphome-solax-x1-mini/assets/95611904/903e3513-ae1a-4828-b8ba-2988525c14f0)
[logs_esphome-web-afe0c6_run(1).txt](https://github.com/syssi/esphome-solax-x1-mini/files/11866042/logs_esphome-web-afe0c6_run.1.txt)

**Reply by etf3k** [reply]
Changing Pins back to 4 and 5 
[logs_esphome-web-afe0c6_run(2).txt](https://github.com/syssi/esphome-solax-x1-mini/files/11866063/logs_esphome-web-afe0c6_run.2.txt)

I tested Pin 3 and 7  as GND on the TX board

Must the Export Control Function activated in Menu for signal on RS485? 
![1687769255833](https://github.com/syssi/esphome-solax-x1-mini/assets/95611904/0f03c52d-5d74-4952-9dc6-82cef231b389)

**Reply by syssi** [reply]
1. Please provide a photo which shows the wiring between the ESP and the RS485 module. I would like to make sure you are using GPIO4/GPIO5 at the ESP side for sure.
2. Please provide a photo of the wiring between the RS485 module and the RJ45 jack. 
3. If you want to monitor the inverter the "Export Control" feature / meter mode must not be enabled!
4. Could you tell me the firmware version of your inverter?

The firmware of my Solax X1 mini doesn't show any settings regarding RS485 and/or Modbus address at the on-screen menu. The inverter did respond OOTB to requests. If your firmware is different you could try to disable the Modbus support if possible.

**Reply by etf3k** [reply]
![1687775478984](https://github.com/syssi/esphome-solax-x1-mini/assets/95611904/fbf8ab9d-7767-402a-a8ad-563a63decf88)
![1687775668705](https://github.com/syssi/esphome-solax-x1-mini/assets/95611904/be2204a9-1ded-4b98-b575-5350a39becca)


In the menu about is Master 1.08 and Manager 1.07 , Internal Code : A 1.42 D 2.25 
Other version information i didnt find.

**Reply by syssi** [reply, marked as answer]
You have to attach the green & yellow wire of the RS485 module to GPIO4 (`D2`) / GPIO5 (`D1`) of the ESP.

**Reply by syssi** [reply]
The TX LED must blink periodically (once per second). If the RX LED blinks periodically you have two swap the green & yellow wire.

**Reply by etf3k** [reply]
THX men i have data !!!!! :D 
How can i make this to a sensor at the energie graph in Homeassist or how can i find the statistic you post in the code menu ?

**Reply by syssi** [reply]
HA -> Settings -> Dashboards -> Energy dashboard -> Solar panels -> "Add solar production". Please select the `sensor.solax_x1_mini_energy_total` sensor here.

**Reply by syssi** [reply]
Is this your first ESPHome node? Do you know how to add the ESPHome node as device to your Home Assistant setup?

**Reply by etf3k** [reply]
Yeah its my first time with homeassistant and esphome, i found my fault. The integration was it. Now i have the energy table. Big THX.

**Reply by syssi** [reply]
You are welcome! Have fun!
